### ISSUE TITLE
set up system to allow clean changes of classnames

### ISSUE THREAD

**Issue by sunnysideup** [silverstripe/silverstripe-framework#10859]
## Affected Version

All versions

## Description

When a DataObject ClassName is changed then the SS project may not work anymore as there may be unexpected references to the classname in the database tables. 

Unexpected as: not the main ClassName field in the table relating to the class itself, but references in other tables (including Foo_Versions). 

## Proposed solution

Use the concepts from this module: https://github.com/sunnysideup/silverstripe-search-all-tables-and-fields and keep a log of changes made.

**Comment by michalkleiner** [comment]
I believe the `DatabaseAdmin::classname_value_remapping` config already provides this for ORM-created db columns. Anywhere else it's project-level code.

**Comment by GuySartorelli** [comment]
Yeah this functionality definitely already exists. I'd say it should be documented though, I couldn't find anything in the docs when I looked briefly just now.
I'll move this to the developer-docs repo.
